compiler/dyno/lib/CMakeLists.txt

```diff
@@ -50,3 +50,4 @@ target_include_directories(libdyno PUBLIC
                            ${CHPL_MAIN_INCLUDE_DIR}
                            ${CHPL_INCLUDE_DIR})
 target_link_libraries(libdyno PUBLIC ${DYNO_LLVM_LINK_ARGS})
+target_compile_options(libdyno PUBLIC SHELL:$<$<COMPILE_LANGUAGE:CXX>:${DYNO_LLVM_COMP_ARGS}>)
```

Comment:
What is this doing?

Reply:
`libdyno-obj` has this same line above, here is the comment from there.
When I remove it, the build step fails.
```
# SHELL: is needed here to remove unwanted quotes from the list of arguments
# COMPILE_LANGUAGE:CXX is needed here to make sure we only set the flags
# for C++ files, not C files.
```

Reply:
OK. Since we have our cmake code running printchplenv now, I think we can and should control the quoting in a way that cmake understands. (So I think printchplenv cmake mode can output with the right quoting for cmake -- also for multiple element things like command line options I think it should separate the elements with `;` to make it a cmake list).

Reply:
So in other words, I think we should try to remove the SHELL part.

For the COMPILE_LANGUAGE part, we have flags for C code that we should be applying as well. It just doesn't break the build not to do so.

Reply:
IMO it would be reasonable to remove `DYNO_LLVM_COMP_ARGS` and `DYNO_LLVM_LINK_ARGS` entirely and just use `CHPL_HOST_BUNDLED_COMPILE_ARGS` / `CHPL_HOST_SYSTEM_COMPILE_ARGS`.